test/runner/run.ts

```diff
@@ -1,180 +0,0 @@
-  );
-  const set = await Promise.all(
-    list.map(async (obj: any) => {
-      const { input, options } = obj;
-      const result = await generateSourceFile(
-        input,
-        {
-          // Enable stage 3 support (ESNext)
-          next: options.next,
-          // Enable React JSX parsing
-          jsx: options.jsx,
-          // Disable web compatibility
-          disableWebCompat: options.disableWebCompat,
-          // Enables implied strict mode
-          impliedStrict: options.impliedStrict
-        },
-        obj.options.module,
-        obj.options.printer,
-        obj.options.incremental
-      );
-
-      return { obj, result };
-    })
-  );
-
-  console.timeEnd(
-    ColorCodes.GREEN + 'Running ' + ColorCodes.RESET + list.length + ' test cases.' + ColorCodes.yellow + ' Total time'
-  );
-
-  await Promise.all(
-    set.map(async ({ obj, result }: any) => {
-      obj.newoutput.ast = result;
-      obj.newoutput.pretty = printSourceFile(result, {
-        tabWidth: 2,
-        printWidth: 80,
-        useTabs: false,
-        bracketSpacing: true
-      } as any);
-    })
-  );
-}
-
-async function generateSourceFile(
-  input: string,
-  options: Options,
-  isModule: boolean,
-  _printer: boolean,
-  /* TODO */ _isIncremental: boolean
-) {
-  return (isModule ? parseModule : parseScript)(input, options);
-}
-
-export async function generateNewOutput(list: any) {
-  list.forEach((obj: any) => {
-    obj.data = generateOutputBlock(obj.data, obj.newoutput.ast, obj.newoutput.pretty, obj.options);
-  });
-}
-
-export async function writeNewOutput(list: any, update: boolean) {
-  let updated = 0;
-  await Promise.all(
-    list.map((obj: any): any => {
-      const { data, previous, file } = obj;
-      if (data !== previous) {
-        if (update) {
-          ++updated;
-          promiseToWriteFile(file, data);
-        } else {
-          process.exitCode = 1;
-          console.log('Output mismatch for', file);
-          return Promise.resolve();
-        }
-      }
-    })
-  );
-  console.log('Updated', updated, 'files');
-}
-
-function parseTestFile(obj: any): any {
-  const { data, previous } = obj;
-
-  if (!previous) return;
-
-  // find the options
-  const optionsOffset = data.indexOf(Constants.Options);
-  const start1 = data.indexOf(Constants.JsStart, optionsOffset);
-  const end1 = data.indexOf(Constants.JsEnd, optionsOffset);
-
-  // Negative if no opions are set, so we pass a empty obj
-  const options = optionsOffset === -1 ? {} : eval('0||' + data.slice(start1 + Constants.JsStart.length, end1) + '');
-
-  const inputOffset = data.indexOf(Constants.Input);
-  const start = data.indexOf(Constants.JsStart, inputOffset);
-  const end = data.indexOf(Constants.JsEnd, inputOffset);
-  const input = data.slice(start + Constants.JsStart.length, end);
-
-  return { options, input };
-}
-
-function generateOutputBlock(currentOutput: any, ast: any, printed: any, options: any) {
```

Comment:
The new changes for this generating output thing results in insane output on windows. see my main comment below.